aws/resource_aws_ec2_managed_prefix_list_entry_test.go

```diff
@@ -0,0 +1,486 @@
+		if ec2err.Code() != "InvalidPrefixListId.NotFound" {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckAWSEc2ManagedPrefixListExists(n string, prefixList *ec2.ManagedPrefixList, entries *[]*ec2.PrefixListEntry) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No EC2 Managed Prefix List is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).ec2conn
+		req := &ec2.DescribeManagedPrefixListsInput{
+			PrefixListIds: []*string{aws.String(rs.Primary.ID)},
+		}
+		resp, err := conn.DescribeManagedPrefixLists(req)
+		if err != nil {
+			return err
+		}
+
+		if len(resp.PrefixLists) > 0 && *resp.PrefixLists[0].PrefixListId == rs.Primary.ID {
+			*prefixList = *resp.PrefixLists[0]
+			input := &ec2.GetManagedPrefixListEntriesInput{
+				PrefixListId: prefixList.PrefixListId,
+			}
+			remoteEntries, err := getEc2ManagedPrefixListEntries(conn, input)
+			if err != nil {
+				return err
+			}
+			*entries = remoteEntries
+			log.Printf("[DEBUG] [within-tests] Entries are : %v", entries)
+			return nil
+		}
+
+		return fmt.Errorf("EC2 Managed Prefix List Entry not found")
+	}
+}
+
+func testAccCheckAWSEc2ManagedPrefixListEntryAttributes(n string, entries *[]*ec2.PrefixListEntry, entry *ec2.PrefixListEntry) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("EC2 Managed Prefix List Entry Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No EC2 Managed Prefix List Entry is set")
+		}
+
+		if entry == nil {
+			entry = &ec2.PrefixListEntry{
+				Cidr: aws.String("10.0.0.0/8"),
+			}
+		}
+
+		var matchingEntry *ec2.PrefixListEntry
+		log.Printf("[DEBUG] [within-tests] Entries are : %v", entries)
+		if len(*entries) == 0 {
+			return fmt.Errorf("No Entries")
+		}
+
+		for _, r := range *entries {
+			if entry.Cidr != nil && r.Cidr != nil && *entry.Cidr != *r.Cidr {
+				continue
+			}
+
+			matchingEntry = r
+		}
+
+		if matchingEntry != nil {
+			log.Printf("[DEBUG] Matching entry found : %s", matchingEntry)
+			return nil
+		}
+
+		return fmt.Errorf("Error here\n\tlooking for %s, wasn't found in %s", entry, entries)
+	}
+}
+
+func testAccAwsEc2ManagedPrefixListEntryImportStateIdFunc(resourceName string) resource.ImportStateIdFunc {
+	return func(s *terraform.State) (string, error) {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return "", fmt.Errorf("not found: %s", resourceName)
+		}
+
+		plID := rs.Primary.Attributes["prefix_list_id"]
+		cidrBlock := rs.Primary.Attributes["cidr_block"]
+
+		var parts []string
+		parts = append(parts, plID)
+		parts = append(parts, cidrBlock)
+
+		return strings.Join(parts, "_"), nil
+	}
+}
+
+func testAccAwsEc2ManagedPrefixListEntryIpv4Config(rInt int) string {
```

Comment:
```suggestion
func testAccAwsEc2ManagedPrefixListEntryIpv4Config(rName string) string {
```

Here and for configs below, change `rInt int` to `rName string` and adjust configs accordingly.